apps/internal/mock/mock.go

```diff
@@ -62,10 +63,13 @@ func WithHTTPStatusCode(statusCode int) responseOption {
 
 // Client is a mock HTTP client that returns a sequence of responses. Use AppendResponse to specify the sequence.
 type Client struct {
+	mu   sync.Mutex
```

Comment:
A field with a Mutex value (i.e. `sync.Mutex` not `*sync.Mutex`) is safe only when instances of the struct are never copied because copying the struct would copy the Mutex but not its internal state i.e., its lockedness. This can create subtle bugs. For example, it means that methods with value receivers wouldn't synchronize with other methods. Client has no such methods and I doubt it's copied anywhere because that would be weird, but this type now comes with a footgun that deserves a comment like "Clients must never be copied because they have a Mutex value"

Reply:
Updating this.